feat: implement letter create message handler (#47)
Garaio-REM/remcloud-message-bus-api: letter_context.md

# Letter Context

## Events

| Type                                          | GARAIO REM         | REM | Description                                                |
|-----------------------------------------------| ------------------ | --- |------------------------------------------------------------|
| [Letters.Letter.Create](#letterslettercreate) | :white_check_mark: | :x: | Stores a letter in GREM to send an invitation for a meeting|

### Letters.Letter.Create

| Field                            | Type      | Content / Remarks                                                                       |
|----------------------------------|-----------|-----------------------------------------------------------------------------------------|
| `eventType`                      | `string`  | `Letters.Letter.Create`                                                                 |
| `data`                           | `hash`    |                                                                                         |

> | &nbsp;&nbsp;`briefVorlage`       | `string`  | template used for generating the letter, Reference of the letter template, **required** |
> | &nbsp;&nbsp;`verwaltungseinheit` | `string`  | reference of the administrative unit, **required**                                      |
> | &nbsp;&nbsp;`sachbearbeiter`     | `string`  | username of the responsible person, **required**                                        |
> | &nbsp;&nbsp;`stichtag`           | `string`  | effective or due date of the letter, **required**                                       |
> | &nbsp;&nbsp;`beschreibung`       | `string`  | subject or short description of the letter, **optional**                                |
> | &nbsp;&nbsp;`unterschriftLinks`  | `string`  | username for the left-side signature block , **optional**                               |
> | &nbsp;&nbsp;`unterschriftRechts` | `string`  | username for the right-side signature block, **optional**                               |

Reviewer: ```suggestion
| &nbsp;&nbsp;`letterTemplateReference`       | `string`  | template reference used for generating the letter, Reference of the letter template, **required** |
| &nbsp;&nbsp;`propertyReference` | `string`  | reference of the property, **required**                                      |
| &nbsp;&nbsp;`clerkUsername`     | `string`  | Username of clerk (Sachbearbeiter) to assign, **required**                                        |
| &nbsp;&nbsp;`dueDate`           | `string`  | effective or due date of the letter, **required**                                       |
| &nbsp;&nbsp;`description`       | `string`  | subject or short description of the letter, **optional**                                |
| &nbsp;&nbsp;`signatureLeft`  | `string`  | username for the left-side signature block , **optional**                               |
| &nbsp;&nbsp;`signatureRight` | `string`  | username for the right-side signature block, **optional**                               |
```

#### Example

```json
{
  "eventType":"Letters.Letter.Create",
  "data": {
    "letterTemplateReference": "liegenschaft_blanko",
    "clerkUsername": "admgar",
    "dueDate": "24.07.2025",
    "propertyReference": "13000",
    "description": "Serial Condominium Letter ONE UNIT",
    "signatureLeft": "welldev",
    "signatureRight": "garaio.dev"
  }
}
```

##### accepted response message

```json
{
  "eventType": "Letters.Letter.CreateAccepted", 
  "data": {
    "reference": 212
  }
}
```

| Field       | Type     | Content / Remarks                          |
| ----------- | -------- |--------------------------------------------|

> | `reference` | `string` | the reference of the newly created letter. |

Reviewer: ```suggestion
| `id` | `string` | the id of the newly created letter. |
```

Reviewer 2: Why exposing the id? We never expose **ids** in messages, always `references`, except for custom tables where we cannot assume a reference exists.

Reviewer: Yes but a letter does not have a reference. So which reference should we return?

##### rejected response message

```json
{
  "eventType": "Letters.Letter.CreateRejected",
  "data": {

>     "reference": null,

Reviewer: ```suggestion
```
Does it make sense to return here something? It should be the id or nothing.
